apps/snitch_api/lib/snitch_api_web/views/product_view.ex

```diff
@@ -37,6 +38,14 @@ defmodule SnitchApiWeb.ProductView do
     product.selling_price |> to_string
   end
 
+  def default_image(product, _conn) do
+    product = product |> Repo.preload(:images)
```

Comment:
Can the default image be preloaded directly using query instead of preloading all images.

Reply:
Ohh yes, conditional preloading for images can be done in this case.